File: fixtures/scale/dart/large/lib/ciphers/xorshift/xorshift_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'xorshift_cipher.dart';
import 'xorshift_key.dart';
import 'xorshift_vectors.dart';

class XorShiftSuite implements CipherSuite {
  @override
  String name() => 'xorshift';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = XorShiftCipher(XorShiftKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in xorshiftSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/xorshift/XorShiftSuite.kt
package cryptobench.ciphers.xorshift

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class XorShiftSuite : CipherSuite {
    override fun name(): String = "xorshift"

    override fun run(): SuiteResult {
        val cipher: Cipher = XorShiftCipher(XorShiftKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in XorShiftVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
